hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestS3GatewayAuditLog.java

```diff
@@ -101,7 +101,7 @@ public static void tearDown() {
   public void testHeadBucket() throws Exception {
     bucketEndpoint.head(bucketName);
     String expected = "INFO  | S3GAudit | ? | user=null | ip=null | " +
-        "op=HEAD_BUCKET {bucket=bucket} | ret=SUCCESS";
+        "op=HEAD_BUCKET {} | ret=SUCCESS";
```

Comment:
This seems to be loss of information (name of bucket (also key name for the `HEAD_KEY`)).

Reply:
This is because the context is null in unit test. So the details can not be shown here.
I added the manual test result in the description, the details are normal.

Reply:
Thanks for the info.  I wonder if it's possible to set the context for unit test.  If the unit test has too weak assertions, and an important part of functionality needs to be verified manually, I think we have a problem.  We should prevent changes that might accidentally break it.

Reply:
+1

Reply:
@adoroszlai @kerneltime Thanks for the review.
Have brought back the detail message in audit log. Since set the context would need to add new dependency to use MultiValueMapImpl, so I chose to use override to include the detail params.
Please have a check.